### PR TITLE
linkedlist

### PR DESCRIPTION
**Pull request by ErinZhang1998**
added linkedlist.wyv to stdlib/wyvern/collections, and it is fully tested. 
added a RandomWrapper.java that can be used to generated random list.
added pair.wyv to stdlib/wyvern. 

Should I put my tests for linkedlist into the examples directory too?

### CODE REVIEW

File: tools/src/wyvern/stdlib/support/RandomWrapper.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package wyvern.stdlib.support;
+import java.util.Random;
+
+public class RandomWrapper {
```
Comment (JonathanAldrich) [inline tools/src/wyvern/stdlib/support/RandomWrapper.java]: Is this needed?  If not, please omit it from the PR.  It doesn't seem to be used.

File: stdlib/wyvern/pair.wyv [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module pair
+
+type Pair
+	type E
+	type F
+	val t1:this.E
```
Comment (JonathanAldrich) [inline stdlib/wyvern/pair.wyv]: Can we rename t1 and t2 "fst" and "snd"?

**Comment by potanin** [comment]
Is there a coherent overall plan of which interfaces for collections will be important (e.g. Set) and which will be implementations provided and whether any sort of inheritance or subtype structure is planned?

**Comment by JonathanAldrich** [comment]
Eventually we'll definitely have Set, and we have tried to make the Collection interface generic to plan for that--but one thing at a time.  List is the most common data structure so we're adding it first.

File: tools/src/wyvern/stdlib/support/RandomWrapper.java (hunk 1 above)
Comment (ErinZhang1998) [reply tools/src/wyvern/stdlib/support/RandomWrapper.java]: Hi Professor Aldrich

My linkedlist tests uses this. 

Do you want me to omit it and pull again?

Thanks!

Comment (JonathanAldrich) [reply tools/src/wyvern/stdlib/support/RandomWrapper.java]: I see.  But your linkedlist tests are not part of the PR right now, correct?  So you should either add the tests to the PR, or remove the RandomWrapper from the PR.

**Pull request merged**